packages/api-contract/src/Abi/index.ts

```diff
@@ -162,9 +162,18 @@ export class Abi {
   /**
    * Warning: Unstable API, bound to change
    */
-  public decodeEvent (data: Bytes | Uint8Array): DecodedEvent {
+  public decodeEvent (data: Bytes | Uint8Array, topic: EventRecord['topics'][0]): DecodedEvent {
```

Comment:
The type for `EventRecord['topics'][0]` should be a `Hash`?

Reply:
Afaik, yes it's a hash. Both should work here. But yes if the structure of the `topic` field of an `EventRecord` gets more complex this breaks. Probably it won't.

Using `Hash` here and renaming the parameter `signature_topic` would make it more atomic and clearer what's happening. Will adapt the code. 👍